gobblin-runtime/src/main/java/org/apache/gobblin/runtime/TaskState.java

```diff
@@ -103,6 +106,7 @@ public TaskState(WorkUnitState workUnitState) {
     this.jobId = workUnitState.getProp(ConfigurationKeys.JOB_ID_KEY);
     this.taskId = workUnitState.getProp(ConfigurationKeys.TASK_ID_KEY);
     this.taskKey = workUnitState.getProp(ConfigurationKeys.TASK_KEY_KEY, "unknown_task_key");
+    this.ignoreCloseFailures = workUnitState.getPropAsBoolean(ConfigurationKeys.TASK_IGNORE_CLOSE_FAILURES, true);
```

Comment:
I think it is safer to default to `false`. The existing behavior can result in silent data loss where the state moves forward on error. Let's consider that behavior a bug and conservatively raise an error. If users encounter an error and decide that ignoring close errors is appropriate then they can update their jobs with this config set to true.

Reply:
Agreed ! Changed it.